packages/@uppy/dashboard/src/components/AddFiles.tsx

```diff
@@ -1,37 +1,53 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-nocheck Typing this file requires more work, skipping it to unblock the rest of the transition.
-
 /* eslint-disable react/destructuring-assignment */
 import { h, Component, Fragment, type ComponentChild } from 'preact'
+import type { I18n } from '@uppy/utils/lib/Translator'
+import type Translator from '@uppy/utils/lib/Translator'
+import type { DashboardState, TargetWithRender } from '../Dashboard.js'
+
+interface AddFilesProps {
+  i18n: I18n
+  i18nArray: Translator['translateArray']
+  acquirers: TargetWithRender[]
+  handleInputChange: (event: Event) => void
+  maxNumberOfFiles: number | null
+  allowedFileTypes: string[] | null
+  showNativePhotoCameraButton: boolean
+  showNativeVideoCameraButton: boolean
+  nativeCameraFacingMode: 'user' | 'environment' | ''
+  showPanel: (id: string) => void
+  activePickerPanel: DashboardState<any, any>['activePickerPanel']
+  disableLocalFiles: boolean
+  fileManagerSelectionType: string
+  note: string | null
+  proudlyDisplayPoweredByUppy: boolean
+}
 
-type $TSFixMe = any
-
-class AddFiles extends Component {
-  fileInput: $TSFixMe
+class AddFiles extends Component<AddFilesProps> {
+  fileInput: HTMLInputElement | null = null
 
-  folderInput: $TSFixMe
+  folderInput: HTMLInputElement | null = null
```

Comment:
```suggestion
  folderInput!: HTMLInputElement
```

And let's remove the optional chaining below

Reply:
🤔 why
I understand `ref` is unlilkely to be `null` when `triggerFileInputClick()` is called, but it is certainly `null` by default. So, if we want to say "we know this is not null now", we should instead do `this.fileInput!.click()` .

But I wouldn't do that either, I think we shouldn't rely on react rendering cycles too much.

Reply:
To reduce the output diff

Reply:
I mean - reducing the output diff is clearly not the value in itself, and we certainly shouldn't `!` values in order to reduce the output diff.

I think your logic is "when I'm viewing the git history of the file, I should be able to ignore typescript commits, because they are highly unlikely to introduce real changes"?
If so - would you like me to subdivide this PR into some chunks, especially for changes that do change output js?

Reply:
I guess there are several points for which it makes sense to me to keep the output diff as close to 0:

- I believe there is value in not artificially increasing the output files size when (IMO in this case) it does not bring any value to our users, and very little to ourselves.
- It makes reviewing so much easier.
- As you said, it makes git blame crawling easier because the TS commit can be safely discarded.

Now I don't feel strongly about either of those points, feel free to ignore, but if you think it does make sense, I'm happy to help split the PR so we can land the uncontroversial TS-only changes first.